zerver/lib/url_preview/preview.py

```diff
@@ -0,0 +1,63 @@
+import re
+import logging
+import traceback
+from six import text_type
+from typing import Any, Optional
+from typing.re import Match
+import requests
+from django.conf import settings
+from zerver.lib.cache import cache_with_key, get_cache_with_key
+from zerver.lib.bugdown import testing_mocks
+from .oembed import get_oembed_data
+from .parsers import OpenGraphParser, GenericParser
+
+
+CACHE_NAME = "database"
+link_regex = re.compile(
+    r'^(?:http)s?://'  # http:// or https://
+    r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|'  # domain...
+    r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'  # ...or ip
+    r'(?::\d+)?'  # optional port
+    r'(?:/?|[/?]\S+)$', re.IGNORECASE)
```

Comment:
This regexp is a modified version of the django.core.validators.URLValidator. Maybe it's better to use original version instead of it? What do you think?